modules/config/default/config.el

```diff
@@ -480,20 +478,27 @@ Continues comments if executed from a commented line. Consults
          "C-S-p" #'corfu-popupinfo-scroll-down
          "C-S-n" #'corfu-popupinfo-scroll-up
          "C-S-u" (cmd! (funcall-interactively #'corfu-popupinfo-scroll-down corfu-popupinfo-min-height))
-         "C-S-d" (cmd! (funcall-interactively #'corfu-popupinfo-scroll-up corfu-popupinfo-min-height)))
-        (:map corfu-map
-         :gi "C-<return>" '(menu-item "Conclude the minibuffer" exit-minibuffer
-                            :enable (active-minibuffer-window))
-         :gi "S-<return>" '(menu-item "Insert completion and conclude" +corfu-complete-and-exit-minibuffer
-                            :enable (active-minibuffer-window))))
+         "C-S-d" (cmd! (funcall-interactively #'corfu-popupinfo-scroll-up corfu-popupinfo-min-height))))
   (when-let ((cmds-del (and (modulep! :completion corfu +tng)
                             (cmds! (and (> corfu--index -1)
                                         (eq corfu-preview-current 'insert))
-                                   #'corfu-reset))))
+                                   #'corfu-reset)))
+             (cmds-ret '(menu-item "Insert completion" corfu-insert
+                         :filter (lambda (cmd)
+                                   (if (eq corfu--index -1)
+                                       (corfu-quit)
+                                     cmd))))
```

Comment:
What's the purpose of this? If you want RET to leave the text range as-is when you don't move the selection (meaning it only completes with a candidate if `corfu-next`/`corfu-previous` got called at some point), this should already happen in `+tng`, because `corfu-preselect` is set to `prompt` there (this means `corfu--index` starts out negative instead of 0). See the definition of `corfu-insert`, it already calls `corfu-quit` for `corfu--index` < 0.

Reply:
Basically the purpose of that is so that RET quits the corfu completion AND passes through to the underlying keybinding if the prompt is selected. I hate having to hit RET twice everywhere because of the popup. This is more of an issue for people that set `corfu-auto-delay` lower than the default.